lumigator/python/mzai/backend/backend/services/jobs.py

```diff
@@ -249,7 +249,7 @@ def _validate_config(self, job_type: str, config_template: str, config_params: d
         else:
             loguru.logger.info(f"Validation for job type {job_type} not yet supported.")
 
-    def _get_job_params(self, job_type: str, record, request: BaseModel) -> dict:
+    def _get_job_params(self, job_type: str, record, request: JobCreate) -> dict:
```

Comment:
Can the logic of this function be extracted out of the jobs service too? idk, maybe some sort of computed property inside each Job type?

Reply:
I'd make a plain function returning or mutating some job_params pre-filled with values from the request itself (or maybe the other way around, but that's not so important right now). And yes, that's one of the purposes of this whole change. It will also make the config_templates unnecessary, but we may still need to take the model into account when mapping things around.

These functions can be attached to the job config class or the job module. I'll make them static functions in the job config class for the moment, but I think the module works better as a namespace.

Reply:
I have placed the equivalent logic in `generate_inference_job_config` and `generate_evaluation_lite_job_config`. They cannot be made purely computed properties of the model, because we have dependencies on the data record, the dataset path, and the storage path, plus some small logic getting the API URL for external services. Let's forget about this detail right now, and address where to put these functions and how to best pass the extra params in a separate follow up PR.